pkg/plugins/diff/service.go

```diff
@@ -142,14 +146,20 @@ func NewService(diffProps map[string]config.DiffProps, pluginProps map[string]co
 	return service, service.Close
 }
 
-func registerPlugins(service *Service, diffProps map[string]config.DiffProps, pluginProps map[string]config.PluginProps) {
+func registerPlugins(service *Service, diffProps map[string]config.DiffProps, pluginProps config.Plugins) {
+	registerDefaultPlugins(service, pluginProps.DefaultPath)
 	for n, p := range diffProps {
 		pluginName := p.PluginName
 		// If the requested plugin wasn't configured with a path, it will be defined under the default location
-		pluginPath := config.DefaultPluginLocation(pluginName)
+		pluginPath := filepath.Join(diffPluginsDefaultPath(pluginProps.DefaultPath), pluginName)
 		pluginVersion := 1 // default version
-		if props, ok := pluginProps[pluginName]; ok {
-			pluginPath = props.Path
+		if props, ok := pluginProps.Properties[pluginName]; ok {
+			pp, err := homedir.Expand(props.Path)
+			if err != nil {
+				logging.Default().Errorf("failed to register a plugin for an invalid path: '%s'", props.Path)
+				continue
+			}
+			pluginPath = pp
 			if props.Version != nil {
```

Comment:
I think we can remove the Version pointer from the configuration as we fallback to version 1 in the code - we can document that 0 is the default.